bin/update-ips.bash

```diff
@@ -1,113 +1,149 @@
+yq_eval() {
+  local config_file="${1}"
+  local config_option="${2}"
+  local expression="${3}"
 
-# Usage: diff_cidrs <config_key> <config_file> <cidrs>...
-# Compares the given list of cidrs with the cidrs configured in the config (with implicit diff return code).
-# If DRY_RUN is set it will output to stdout, else to null
-diff_cidrs() {
-  local config_key="${1}"
-  local config_file="${2}"
-  local cidrs=("${@:3}")
+  local config_filename="${config_file//${CK8S_CONFIG_PATH}\//}"
 
-  if $DRY_RUN; then
-    out_file=/dev/stdout
+  local out
+  if ${dry_run}; then
+    out=/dev/stdout
   else
-    out_file=/dev/null
+    out=/dev/null
   fi
 
-  # use diff return implicitly
   diff -U3 --color=always \
-    --label "${config_file//${CK8S_CONFIG_PATH}\//}" <(yq4 -P "${config_key}"' // [] | sort' "${config_file}") \
-    --label expected <(yq4 -P 'split(" ") | sort' <<< "${cidrs[*]}") > "${out_file}"
+      --label "${config_filename}" <(yq4 -P "${config_file}") \
+      --label expected <(yq4 -P "${expression}" "${config_file}") > "${out}" && return
+
+  if ${dry_run}; then
+    log_warning "Diff found for ${config_option} in ${config_filename} (diff shows actions needed to be up to date)"
+  else
+    yq4 -i "${expression}" "${config_file}"
+  fi
+
+  has_diff=$((has_diff + 1))
 }
 
-# Usage: update_cidrs <config key> <config file> <cidrs>
-update_cidrs() {
-  local config_key="${1}"
-  local config_file="${2}"
-  local cidrs=("${@:3}")
+# Set the value of a list configuration option.
+#
+# Usage: yq_set_list <config_file> <expression> <list-item> [<list-item> ...]
+yq_set_list() {
+  local config_file="${1}"
+  local config_option="${2}"
+  shift 2
 
-  local ips
-  ips="$(yq4 -oj 'split(" ") | sort' <<< "${cidrs[*]}")"
+  local list
+  list=$(echo "[$(for v in "${@}"; do echo "${v},"; done)]" | yq4 -oj sort)
 
-  yq4 -i "${config_key} = ${ips}" "${config_file}"
+  yq_eval "${config_file}" "${config_option}" "${config_option} = ${list}"
 }
 
-# Fetches the IPs from a specified address
-# Usage: getDNSIPs <dns_record>
-getDNSIPs() {
-  local IPS
-  mapfile -t IPS < <(dig +short "${1}" | grep '^[.0-9]*$')
-  if [ ${#IPS[@]} -eq 0 ]; then
-    log_error "No ips for ${1} was found"
+# Fetch the IPs from a domain.
+#
+# Usage: get_dns_ips <domain>
+get_dns_ips() {
+  local domain="${1}"
+
+  local -a ips
+  mapfile -t ips < <(dig +short "${domain}" | grep '^[.0-9]*$')
+  if [ ${#ips[@]} -eq 0 ]; then
+    log_error "No ips for ${domain} was found"
     exit 1
   fi
-  echo "${IPS[@]}"
+  echo "${ips[@]}"
 }
 
-# Fetches the Internal IP and calico tunnel ip of kubernetes nodes using the label selector.
-# If label selector isn't specified, all nodes will be returned.
-getKubectlIPs() {
-  local IPS_internal
-  local IPS_calico
-  local IPS
+# Fetches the InternalIP, Calico tunnel IP and Wireguard IP of Kubernetes
+# nodes using the label selector. If label selector isn't specified, all
+# nodes will be returned.
```

Comment:
```suggestion
# Fetches the InternalIP, Calico tunnel IP and Wireguard IP of Kubernetes nodes using the label selector.
# If label selector isn't specified, all nodes will be returned.
```

Reply:
I have a habit of cutting of comments at 80 characters. Do we have any guidelines that says this is incorrect or why are you suggesting this?

Reply:
for me it just feels like 80 characters is a bit low to have a whole sentence just cut off at while continuing on the other line  at least with something in between like the `#`.
but otherwise without my feelings in the way it can at least be good to start a new sentence in a new line like this so its clearly marked that it is a new sentence.
```suggestion
# Fetches the InternalIP, Calico tunnel IP and Wireguard IP of Kubernetes
# nodes using the label selector. 
# If label selector isn't specified, all nodes will be returned.
```

Reply:
As long as we don't have a guideline saying otherwise I'll stick to 80 chars. I would not be against having a guideline though.

I think your suggestion to add some spacing to the comments make sense though https://github.com/elastisys/compliantkubernetes-apps/pull/1853/commits/489d45fe3b9443137528947e74c6742dacae2b7d